src/connection.rs

```diff
@@ -207,4 +217,59 @@ impl Connection {
             },
         }
     }
+
+    fn send_query(&self, tree: &Tree) -> Result<(), Error> {
+        let token = self.query_token.get();
+        // Increment the token for the next request.
+        self.query_token.set(token.wrapping_add(1));
```

Comment:
I think there are two techniques for setting the wrapping attribute in rust. One is a function as you've done here and the other is declaring it in the definition. I prefer the second as this is required everywhere IMO. 

https://doc.rust-lang.org/1.6.0/std/num/struct.Wrapping.html